### ISSUE TITLE
Refactoring CSS Pre-Processing

### ISSUE THREAD

**Issue by haydenbleasel** [yeoman/generator-webapp_DEPRECATED#237]
I've been thinking about the way Yeoman handles stylesheets lately, especially with regards to preprocessing with languages / frameworks like Less, Sass and Compass. I have some thoughts:
#### 1) Make the Compass task optional

I'm not going to discuss this point in detail since there's already a popular issue on [making the Compass task optional](https://github.com/yeoman/generator-webapp/issues/114), but the general proposal is to remove the dependancy on Compass with regards to the second issue...
#### 2) Remove Bootstrap from the Sass option

The current Yeoman scaffold includes an option for Bootstrap for Sass. Selecting this option informs the scaffold uses Sass / Compass as the default stylesheet language (main.css -> main.scss).

My proposal is to remove Bootstrap from the Sass option, so the option is simply "Use Sass pre-processing". If the user wants Bootstrap for Sass, they can get it through Bower on their own accord. This also helps since the user may want to use Sass but not necessarily implement Bootstrap.
#### 3) Include Less as a pre-processor option

The title should be fairly clear, but adding Less as a CSS pre-processing option would be a good idea. I guess options for this are [grunt-contrib-less](https://github.com/gruntjs/grunt-contrib-less) or [grunt-recess](https://github.com/sindresorhus/grunt-recess).
#### TL;DR

Options for using Less and Sass without the need for Bootstrap included and removing the dependancy on Compass.

**Comment by addyosmani** [comment]
I agree that we can improve the current CSS preprocessor setup, however, I personally disagree about removing Bootstrap outright however. When you're starting a new project, especially during prototyping, having a set of default styles out of the box can be pretty powerful. Perhaps break Sass/Compass/Bootstrap out into prompts that make sense rather than dropping features?

**Comment by sindresorhus** [comment]
What Addy said. This generator is meant to be an opinionated starting point for creating web apps.

Maybe prompts like:
- Sass
- Bootstrap
- Modernizr

## 

Re LESS, see https://github.com/yeoman/yeoman/issues/289 We prefer Sass, but it should be easy to switch that out with LESS.

**Comment by haydenbleasel** [comment]
I'm totally for Addy's idea of breaking up Sass / Compass / Bootstrap into options. I just think it's wrong bundling Bootstrap with Sass by default. I totally agree with you guys, Bootstrap is a perfect starting point for a web application, however Yeoman should be providing the choice to start with Bootstrap, rather than forcing the user to remove it if they don't need it.

In a way, removing the necessity of Bootstrap introduces Yeoman to multiple parts of the workflow. For instance, if you've already done designs and mockups externally and you're ready to write the CSS from ground-up (for smaller web applications that don't require Bootstrap), you simply opt not to include Bootstrap.

With regards to the prompt, I'm happy with Sindre's one for now, however it's not logical if we eventually include Less as an option.

**Comment by sindresorhus** [comment]
I think we'll go with:

```
What do you want?

- Sass?
- Bootstrap? (if yes on the previous, you'll get bootstrap sass)
- Modernizr?
```

Compass will be gone.

---

> With regards to the prompt, I'm happy with Sindre's one for now, however it's not logical if we eventually include Less as an option.

We won't. If you want LESS feel free to fork it and publish one with LESS instead ;)

**Comment by haydenbleasel** [comment]
@sindresorhus - Yeah that totally works. As a result, gives the users who don't want Sass the option to include Bootstrap by default too. Did you want me to work on this and submit a PR? In regards to the whole Less thing, I reckon I'll create my own distribution of Yeoman like you suggested.

**Comment by sindresorhus** [comment]
@haydenbleasel sure! though i would prefer if we removed Compass first (happy to accept a PR on that too).

**Comment by haydenbleasel** [comment]
@sindresorhus - Replace Compass with grunt-sass as discussed in [this issue](https://github.com/yeoman/generator-webapp/issues/114) and binding it to the Sass option?

**Comment by sindresorhus** [comment]
@haydenbleasel nah, libsass (and then grunt-sass) is still way too unstable to be default.

**Comment by haydenbleasel** [comment]
@sindresorhus - No dramas. I'll just bind Compass to the new Sass option then.

**Comment by sindresorhus** [comment]
Replace grunt-contrib-compass with grunt-contrib-sass

**Comment by haydenbleasel** [comment]
@sindresorhus - I'm not sure replacing Compass with Sass is going to be easy, we're currently using a lot of grunt-contrib-compass options that don't exist in grunt-contrib-sass e.g.
- generatedImagesDir
- imagesDir
- javascriptsDir
- fontsDir
- importPath
- httpImagesPath
- httpGeneratedImagesPath
- httpFontsPath
- relativeAssets
- assetCacheBuster

**Comment by haydenbleasel** [comment]
@addyosmani - Also contributes to [your issue](https://github.com/yeoman/generator-webapp/issues/168).

**Comment by addyosmani** [comment]
Thanks! I've commented on your PR. It looks like @sindresorhus is down for this set of changes. Once we've reviewed your commits and there are no oppositions, we'll land this :)
